polaris-react/src/components/DropZone/DropZone.stories.tsx

```diff
@@ -533,6 +536,108 @@ export function WithCustomFileDialogTrigger() {
   );
 }
 
+export function WithCustomModalAndDragAndDropOnly() {
+  const [files, setFiles] = useState([]);
+  const [activeModal, setActiveModal] = useState(false);
+
+  const handleDrop = useCallback((dropFiles) => {
+    setFiles((files) => [...files, dropFiles]);
+  }, []);
+
+  const handleOpenModal = useCallback(
+    () =>
+      setActiveModal((activeModal) => {
+        console.log(activeModal);
+        return !activeModal;
+      }),
+    [],
+  );
+
+  const handleOpenModalFromChildElement = useCallback(
+    () => (event) => {
+      event.stopPropagation();
+      setActiveModal((activeModal) => !activeModal);
+    },
+    [],
+  );
+
+  const validImageTypes = ['image/gif', 'image/jpeg', 'image/png'];
+  const fileUpload = !files.length && (
+    <UnstyledButton
+      onClick={handleOpenModalFromChildElement}
+      style={{width: 40, height: 40, border: 0, cursor: 'pointer'}}
+      accessibilityLabel="Upload image for product Green T-shirt size Small"
+    >
+      <Icon source={ImageAddIcon} tone="interactive" />
+    </UnstyledButton>
+  );
+
+  const uploadImageModal = (
+    <div style={{height: '500px'}}>
+      <Modal
+        size="small"
+        open={activeModal}
+        onClose={handleOpenModal}
+        title="Upload images from store gallery"
+        primaryAction={{
+          content: 'Save',
+          onAction: handleOpenModal,
+        }}
+        secondaryActions={[
+          {
+            content: 'Cancel',
+            onAction: handleOpenModal,
+          },
+        ]}
+      >
+        <Modal.Section>
+          <DropZone accept=".png" type="file" onDrop={() => {}}>
+            <DropZone.FileUpload />
+          </DropZone>
+        </Modal.Section>
+      </Modal>
+    </div>
+  );
+
+  const uploadedFiles = files.length > 0 && (
+    <BlockStack gap="400">
+      {files.map((file, index) => (
+        <InlineStack gap="400" align="center" key={index}>
+          <Thumbnail
+            size="small"
+            alt={file.name}
+            source={
+              validImageTypes.includes(file.type)
+                ? window.URL.createObjectURL(file)
+                : NoteIcon
+            }
+          />
+        </InlineStack>
+      ))}
+    </BlockStack>
+  );
+
+  return (
+    <BlockStack gap="200">
+      <Text as="h2" variant="headingMd">
+        Small sized Drop zone that functions as a drop zone when using a mouse
+        and has a customized modal that opens when clicked. When navigating with
+        a keyboard, the child element serves as the modal activator and provides
+        accessibility and custom desight features.
```

Comment:
Is this intentional?
```suggestion
        accessibility and custom features.
```